File: 200_LeetCode_Questions/D01P2_LeetCode_1470.java
//Runtime: 0 ms, faster than 100.00% of Java online submissions for Shuffle the Array.
//Memory Usage: 39.8 MB, less than 59.61% of Java online submissions for Shuffle the Array.

class Solution {
    public int[] shuffle(int[] nums, int n) {
        int[] shuffled = new int[nums.length];
        for(int i = 0, j = 0; j < n; i+=2, j++){
            shuffled[i] = nums[j];
            shuffled[i + 1] = nums[j + n];
        }
        return shuffled;
    }
}

File: 200_LeetCode_Questions/C++/D01P2_LeetCode_1470.cpp
class Solution {
public:
   vector<int> shuffle(vector<int>& nums, int n) {
       for (int i = 2*n-1; i > 0; i--) {
           int j = i;
           do {
               if (j % 2 == 1)
                   j = j/2 + n;
               else
                   j = j/2;
           } while (j > i);
           swap(nums[i], nums[j]);
       }
       return nums;
   }
};